IEP-1445 GH #1148: Avoid creating '.clang-format' on each compilation

sigmaaa opened the pull request:
## Description

Added an option to disable/enable auto-creation of clang-format file
![image](https://github.com/user-attachments/assets/f0ae0f61-f519-4acb-853d-89cc5597a008)


Fixes # ([IEP-1445](https://jira.espressif.com:8443/browse/IEP-1445))

## Type of change

Please delete options that are not relevant.
- New feature (non-breaking change which adds functionality)

## How has this been tested?
Created/built the project with this option turned off and on, then checked if clang-format was created accordingly

**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):

## Dependent components impacted by this PR:

- clang-format file
- Espressif preference page

## Checklist
- [ ] PR Self Reviewed
- [ ] Applied Code formatting
- [ ] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Added a new option in the preferences that lets users enable or disable the automatic creation of the `.clang-format` file.
	- The new setting includes an informative tooltip and defaults to an enabled state, ensuring a consistent formatting configuration during builds.
	- Introduced new constants for preference management related to Clang format file automation.
- **Localization**
	- Added new strings for button naming and tooltip messages to enhance user interface localization for the preferences page.
- **Documentation**
	- Updated the documentation to include instructions on the automatic creation of `.clang-format` files and how to manage this setting.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

AndriiFilippov commented:
@sigmaaa hi !

Tested under:
OS - Windows 11 
ESP-IDF: v5.4 

Created/built the project with this option turned off and on, and verified that the `.clang-format` file was created only when the option was enabled, confirming the expected behavior. ✅ 

Created `.clang-format` file inside root directory - build project - checked if it able to find `.clang-format` file outside project folder ✅ 
 
 
 LGTM 👍

AndriiFilippov commented:
@kolipakakondal @alirana01  hi ! 

Please, review

alirana01 reviewed (approved):
LGTM just need to confirm with @AndriiFilippov and @kolipakakondal if this is okay in terms of UX
Because I am not sure if this should be under Build settings or a separate editor settings.

kolipakakondal reviewed (changes requested):
Can you add this to the Clangd editor documentation as well? Please explain the reason for doing this and mention that it is useful for projects migrated from earlier versions (2.x to 3.x) to ensure compatibility with the Clangd editor. Otherwise, it looks good to me.

kolipakakondal commented:
> LGTM just need to confirm with @AndriiFilippov and @kolipakakondal if this is okay in terms of UX Because I am not sure if this should be under Build settings or a separate editor settings.

Agree. We are doing this during the build to address old project migrations (2.x to 3.x). We could categorize this under "Clangd Settings" so that, going forward, any additional related settings can be added here.

sigmaaa commented:
> > LGTM just need to confirm with @AndriiFilippov and @kolipakakondal if this is okay in terms of UX Because I am not sure if this should be under Build settings or a separate editor settings.
> 
> Agree. We are doing this during the build to address old project migrations (2.x to 3.x). We could categorize this under "Clangd Settings" so that, going forward, any additional related settings can be added here.

Hi @kolipakakondal, updated the documentation and updated Espressif preference page as follows:
![image](https://github.com/user-attachments/assets/98094a72-0246-4073-bc07-c68583db9b47)

kolipakakondal commented:
LGTM
